src/compiler/checker.ts

```diff
@@ -11896,13 +11896,19 @@ namespace ts {
                     else if ((type as MappedType).objectFlags & ObjectFlags.Mapped) {
                         resolveMappedTypeMembers(type as MappedType);
                     }
+                    else {
+                        Debug.fail("Unhandled object type " + Debug.formatObjectFlags(type.objectFlags));
+                    }
                 }
                 else if (type.flags & TypeFlags.Union) {
                     resolveUnionTypeMembers(type as UnionType);
                 }
                 else if (type.flags & TypeFlags.Intersection) {
                     resolveIntersectionTypeMembers(type as IntersectionType);
                 }
+                else {
+                    Debug.fail("Unhandled type " + Debug.formatTypeFlags(type.flags));
+                }
             }
             return type as ResolvedType;

```

Comment:
Maybe this should assert each of the things that `ResolvedType` is assuming are defined, but I don't know how expensive that would be.